app/products/calls/connection/connection.ts

```diff
@@ -424,15 +525,20 @@ export async function newConnection(
             }
         });
 
-        peer.on('stream', (remoteStream: MediaStream) => {
-            logDebug('calls: new remote stream received', remoteStream.id);
+        peer.on('stream', (remoteStream: MediaStream, trackInfo: TrackInfo) => {
+            logDebug('new remote stream received', remoteStream.id, trackInfo);
             for (const track of remoteStream.getTracks()) {
-                logDebug('calls: remote track', track.id);
+                logDebug('remote track', track.kind, track.id);
             }
 
             streams.push(remoteStream);
+
             if (remoteStream.getVideoTracks().length > 0) {
-                setScreenShareURL(remoteStream.toURL());
+                if (trackInfo?.type === 'video') {
```

Comment:
`trackInfo` would be undefined when connecting to older backends. In that case, we assume any video track to be screen sharing.